consensus/chain_selector.go

```diff
@@ -33,8 +48,62 @@ func NewChainSelector(cs *hamt.CborIpldStore, actorState SnapshotGenerator, gCid
 	}
 }
 
-// Weight returns the expected consensus weight of this TipSet in uint64
-// encoded fixed point representation.
+// NewWeight returns the EC weight of this TipSet in uint64 encoded fixed point
+// representation.
+//
+// w(i) = w(i-1) + (PI)^(P_n) * [V * num_blks + X ]
+// P_n(n) = if n < 3:0 else: n, n is number of null rounds
+// X = log_2(total_storage(pSt))
+func (c *ChainSelector) NewWeight(ctx context.Context, ts types.TipSet, pSt state.Tree) (uint64, error) {
+	ctx = log.Start(ctx, "Expected.Weight")
+	log.LogKV(ctx, "Weight", ts.String())
+	if ts.Len() == 1 && ts.At(0).Cid().Equals(c.genesisCid) {
+		return uint64(0), nil
+	}
```

Comment:
This would be more robust as a function of height, but ok to leave that until we come back to update yet again.